### ISSUE TITLE
🗒 A note on the Dell TB16

### ISSUE THREAD

**Issue by rgov** [rgov/Thunderbolt3Unblocker#28]
Many people file bugs here about their Dell TB16 dock.

The USB, Thunderbolt 3, and audio out ports should regain functionality after installing Thunderbolt 3 Unblocker.

**However, video output does not work.** Unfortunately, this is out of scope for this project.

Thunderbolt 3 Unblocker allows the Thunderbolt 3 driver to recognize the TB16, but other drivers in the system are responsible for activating the downstream peripherals, including the display. [The issue](https://support.displaylink.com/forums/287786-displaylink-feature-suggestions/suggestions/33252664-macos-high-sierra-10-13-4) may be resolved by installing an updated DisplayLink driver.

**Comment by flyingtango** [comment]
I have 2018 new macbook pro with mojave, and the mac can recognize tb15. However, the tb15 can only charge, no usb and no video ouput.

**Comment by barongun** [comment]
@flyingtango  me too....ㅠㅠ

**Comment by Michaelkire** [comment]
@flyingtango & @barongun If you've installed this driver correctly, you should be able to use USB on the tb15. Mine works.

Also I found that I sometimes have to run the install command again, after I have restarted the mac.

You can check in the Apple -> About this Mac -> System report -> Thunderbolt. It should say if its supported or not

**Comment by slvrscoobie** [comment]
Thanks guys, just prevented me from picking up a TB16 on ebay for use with my MBP TB3. I need TB3 video out to my apple display and displayport for my dell display so this was so close, and so cheap but i guess ill have to stick with my 3x more expensive StarTech usb c dock. works perfectly if anyones interested

**Comment by pmtremblay** [comment]
Hey guys,
I have a new Dell: https://www.dell.com/en-ca/shop/dell-laptops-netbooks-and-tablets/new-xps-15/spd/xps-15-9570-laptop/nxps1570_bt_s1649e

Ports & Slots
1. SD card slot | 2. USB 3.1 Gen 1 with Power Delivery | 3. Battery gauge button and indicator | 4. Noble lock slot | 5. AC power | 6. HDMI 2.0 | 7. Thunderbolt™ 3 (4 lanes of PCI Express Gen 3) supporting: Power Delivery, Thunderbolt 3 (40Gbps bi-directional), USB 3.1 Gen 2 (10Gbps), Native DisplayPort 1.2 video output, VGA, HDMI, Ethernet and USB-A via Dell Adapter (sold separately) | 8. Headset jack

I thought I could use my old Apple Cinema display (mini-display port) with my Dell (through USB-C = Thunderbolt3 port) running Mojave. Is there anyway to make my Cinema Display to work with an USB-C hub or otherwise?

Suggestions welcome.

Thanks.

**Comment by laris** [comment]
TB16

- USB2/3 works
- LAN works
- video no work

Hackintosh with DELL P5510 
```
                    'c.          MacBookPro.lan
                 ,xNMM.          -------------------------
               .OMMMMo           OS: macOS Mojave 10.14.1 18B75 x86_64
               OMMM0,            Host: MacBookPro13,3
     .;loddo:' loolloddol;.      Kernel: 18.2.0
   cKMMMMMMMMMMNWMMMMMMMMMM0:    Uptime: 21 mins
 .KMMMMMMMMMMMMMMMMMMMMMMMWd.    Packages: 95 (brew)
 XMMMMMMMMMMMMMMMMMMMMMMMX.      Shell: zsh 5.3
;MMMMMMMMMMMMMMMMMMMMMMMM:       Resolution: 1920x1080@2x
:MMMMMMMMMMMMMMMMMMMMMMMM:       DE: Aqua
.MMMMMMMMMMMMMMMMMMMMMMMMX.      WM: Quartz Compositor
 kMMMMMMMMMMMMMMMMMMMMMMMMWd.    WM Theme: Blue (Dark)
 .XMMMMMMMMMMMMMMMMMMMMMMMMMMk   Terminal: iTerm2
  .XMMMMMMMMMMMMMMMMMMMMMMMMK.   Terminal Font: Monaco 14
    kMMMMMMMMMMMMMMMMMMMMMMd     CPU: Intel i7-6820HQ (8) @ 2.70GHz
     ;KMMMMMMMWXXWMMMMMMMk.      GPU: Intel HD Graphics 530
       .cooc,.    .,coo:.        Memory: 3261MiB / 32768MiB
```

**Comment by orappold** [comment]
TB16 - MacBook Air 2018. Thanks to the unblocked it is recognized, however **video does not** work. 
Any tips, patches, driver update news are always welcome. 

THX

**Comment by pitzor** [comment]
I also have the new macbook air 2018 but the dell tb16 will only charge after applying the script, no video (using the Apple Cinema Display 27), no usb. Had to Modify the script to make it work with 10.4.1. Also looking for a solution...

**Comment by kurikomoe** [comment]
Me too, tb15 is recognized, usb is working but hdmi dp are all not working.

**Comment by rgov** [comment]
I'm locking this topic. **Video does not work on TB16.** That's the message I wrote above. If you try it, and it doesn't work, please don't be surprised. It doesn't work. This project doesn't fix it. No additional data points are necessary.

**Comment by rgov** [comment]
## Update

If you are still interested in a solution, please try [performing a firmware upgrade][firmware] on the TB16, which has been reported to restore video. However, the firmware upgrade must be performed from Windows. It is unlikely that this will work from WINE or a virtual machine.

[firmware]: https://www.dell.com/support/home/us/en/04/drivers/driversdetails?driverId=7TJ1H&osCode=&productCode=dell-thunderbolt-dock-tb16

**Comment by orappold** [comment]
Works for me like a charm - thank you so much for the update!

**Comment by kurikomoe** [comment]
thanks for your update, however I am using the out-of-date tb15 dock. Hoping dell can release some firmware updates

**Comment by NihalPandit** [comment]
Tried Updating the firmware using win10 bootcamp and got "unable to get system info" error. anybody run into this error before..?

**Comment by jaredneedell** [comment]
Just FYI folks

The Dell D600 docks work with the Macbook using DisplayLink software. I have one for work and one for home. Home uses Dual 4k and at work I'm using triple 1080p monitors. No issues plus it charges the macbook. 
https://www.dell.com/en-us/shop/dell-universal-dock-d6000/apd/452-bcyt/pc-accessories

**Comment by jackie099** [comment]
> Tried Updating the firmware using win10 bootcamp and got "unable to get system info" error. anybody run into this error before..?

I am getting the same error. Tried on both macbook pro 2018 amd razer blade 15. I guess we need a xps 15 to update it.
 
I was playing with different TB3 drivers from Intel NUCs and now I am getting "Could not find valid policy or resource payload"

**Comment by c-po** [comment]
Using 
* macOS 10.14.3
* Thunderbolt3Unblocker v0.04
* DELL TB16 Firmware Version 1.0.2

 I can confirm the HDMI output is working as already mentioned. The Display ports are (yet) not working but I can confirm them working with Windows 10.

Connection a second display to a mDP downstream port for MST the second disaply is not recognized.

Anyone tried connecting a second display via USB-C/TB3 downstream port of the DELL TB16?

**Comment by NihalPandit** [comment]
@c-po What computer did you use to update the firmware?

Thanks

**Comment by c-po** [comment]
@NihalPandit I have a DELL 7370, too which is used for FW updating through Windows 10.

I can confirm the Ethernet Port is working, too

**Comment by AndrejKocis** [comment]
> @NihalPandit I have a DELL 7370, too which is used for FW updating through Windows 10.
> 
> I can confirm the Ethernet Port is working, too

Hi mate, whre you find  Thunderbolt3Unblocker version 0.04 ???

**Comment by NihalPandit** [comment]
> > @NihalPandit I have a DELL 7370, too which is used for FW updating through Windows 10.
> > I can confirm the Ethernet Port is working, too
> 
> Hi mate, whre you find Thunderbolt3Unblocker version 0.04 ???
https://github.com/rgov/Thunderbolt3Unblocker/releases/download/v0.0.4/Thunderbolt3Unblocker.zip

**Comment by c-po** [comment]
I was indeed referring to v0.0.4 and not v0.04, sorry!

But I can confirm that the USB-C downstream port on the TB16 can drive one DisplayPort Display, too.

Together with the HDMI output one could connect up to two displays to this Dock. Using Windows one could drive up to 4 displays with the TB3 downstream port

**Comment by pavanagrawal123** [comment]
@jackie099 and @NihalPandit I had that same issue and I was able to resolve it by updating to an older version of the dock firmware and then updating to the newest version. Jumping to the latest from very old firmware doesn't seem to work, but hopping between works.

**Comment by fenneclabs** [comment]
@pavanagrawal123 do you mean you got TB16 displays to work with some firmware combination?

**Comment by pavanagrawal123** [comment]
@fenneclabs Yep I'm able to use the HDMI port on my TB16!

**Comment by fenneclabs** [comment]
Can you give details on which firmware version(s) to use?
I'm using latest version and TB16 does not work.

**Comment by the-wr** [comment]
DP works too, not only HDMI.

**Comment by orappold** [comment]
For me only HDMI works. If I use DP then I get synced screens not extended ones. Weird.

**Comment by the-wr** [comment]
> For me only HDMI works. If I use DP then I get synced screens not extended ones. Weird.

That's weird. I'm able to get extended screen with DP and latest dock firmware. I only connect one external monitor and it's 1080p. Maybe that's the reason.

**Comment by orappold** [comment]
Maybe, I tried 2x 4k external via hdmi and dp.

**Comment by c-po** [comment]
So you have one screen connected via DP and the other via HDMI? Here you can have two extended screens (not synced)?

**Comment by the-wr** [comment]
> So you have one screen connected via DP and the other via HDMI? Here you can have two extended screens (not synced)?

No, unfortunately I can't get the second external display to work separately.
The best outcome I've been able to achieve with MBP so far is to have two external displays (DP+HDMI) mirror each other while being separate from the laptop screen, i.e. two desktops being shown on three screens total.

VGA output doesn't work at all.

**Comment by fenneclabs** [comment]
Then it's useless, I need to connect 3 external screens. Looks like it's only displayport alternate mode that is working.

**Comment by peacechen** [comment]
@pavanagrawal123 
Would you mind providing links to the older TB16 firmwares?  I picked up a used TB16, and the Dell update software on my Inspiron 15 7577 updated the firmware on the TB16.  However OSX won't display video on either the HDMI or DisplayPort outputs (no monitor is detected at all).

When I try the 1.0.2 firmware (the only TB16 firmware I can find searching online), it says no dock detected.  I unplug the dock, remove power, add power, plug it back in as directed.  I suspect the Inspiron 15 isn't in the app's list of supported laptops :(

**Comment by pavanagrawal123** [comment]
@peacechen The only reason I found the old firmware to be helpful was when updating to latest. I wasn't able to update to latest because it was too far of a jump, so if you are able to jump to latest directly I am afraid using an old firmware won't help.

**Comment by csitarichie** [comment]
@peacechen I was using this updater and HDMI works with 3K monitor with MBP 2018 13" 
https://www.dell.com/support/home/us/en/04/drivers/driversdetails?driverid=7tj1h
Make sure that on the laptop where you run windows you have the latest intel thunderbolt controller driver. 
I was using :  17.4.79.510_A12.
Otherwise the Firmware update might fail for the dock.

**Comment by peacechen** [comment]
Thanks for the link to the Thunderbolt Controller driver.  I previously had A10 installed. Updated to A12 but the TB16 firmware updater app still refuses to acknowledge the dock.

The Dell Update center did update the firmware, but didn't indicate which.  Would you check what yours is when connected to a Mac?  In OSX it can be shown going to 
About This Mac -> System Report -> Thunderbolt.

Mine shows fw 16.16 (link firmware 0.25).  It's puzzling how some people are able to get video output, but this one won't output any video.  I wonder if it's the Macbook (mine is also a Pro 13" 2018), or the TB16.

```
Thunderbolt Bus:

  Vendor Name:	Apple Inc.
  Device Name:	MacBook Pro
  UID:	0x0001AF6CC8926200
  Route String:	0
  Firmware Version:	39.2
  Domain UUID:	52A9C074-23F0-EB5F-9A4A-84FD16829106
  Port:
  Status:	No device connected
  Link Status:	0x7
  Speed:	Up to 40 Gb/s x1
  Current Link Width:	0x1
  Receptacle:	2
  Link Controller Firmware Version:	0.38.0
  Port:
  Status:	Device connected
  Link Status:	0x2
  Speed:	Up to 40 Gb/s x1
  Current Link Width:	0x2
  Receptacle:	1
  Link Controller Firmware Version:	0.38.0

Dell Thunderbolt Cable:

  Vendor Name:	Dell
  Device Name:	Dell Thunderbolt Cable
  Vendor ID:	0xD4
  Device ID:	0xB051
  Device Revision:	0x2
  UID:	0x00D44F41C8406800
  Route String:	3
  Firmware Version:	16.16
  Port (Upstream):
  Status:	Device connected
  Link Status:	0x2
  Speed:	Up to 40 Gb/s x1
  Current Link Width:	0x2
  Link Controller Firmware Version:	0.25.0
  Port:
  Status:	Device connected
  Link Status:	0x2
  Speed:	Up to 40 Gb/s x1
  Current Link Width:	0x2
  Link Controller Firmware Version:	0.25.0

Dell Thunderbolt Dock:

  Vendor Name:	Dell
  Device Name:	Dell Thunderbolt Dock
  Vendor ID:	0xD4
  Device ID:	0xB054
  Device Revision:	0x1
  UID:	0x80866E3F953A3210
  Route String:	303
  Firmware Version:	16.16
  Port (Upstream):
  Status:	Device connected
  Link Status:	0x2
  Speed:	Up to 40 Gb/s x1
  Current Link Width:	0x2
  Link Controller Firmware Version:	0.25.0
  Port:
  Status:	No device connected
  Link Status:	0x7
  Speed:	Up to 40 Gb/s x1
  Current Link Width:	0x1
  Link Controller Firmware Version:	0.25.0
```

**Comment by csitarichie** [comment]
@peacechen Thunderbolt Bus 0:

  Vendor Name:	Apple Inc.
  Device Name:	MacBook Pro
  UID:	0x000XXXXXXXXXXXXXXXX
  Route String:	0
  Firmware Version:	34,6
  Domain UUID:	XXXXXXXXXXXXXXXXXXXXXXX
  Port:
  Status:	Device connected
  Link Status:	0x2
  Speed:	Up to 40Gb/s x1
  Current Link Width:	0x2
  Receptacle:	2
  Link Controller Firmware Version:	0.97.0
  Port:
  Status:	No device connected
  Link Status:	0x7
  Speed:	Up to 40Gb/s x1
  Current Link Width:	0x1
  Receptacle:	1
  Link Controller Firmware Version:	0.97.0

Dell Thunderbolt Cable:

  Vendor Name:	Dell
  Device Name:	Dell Thunderbolt Cable
  Vendor ID:	0xD4
  Device ID:	0xB051
  Device Revision:	0x2
  UID:	0x00D4461D6D817600
  Route String:	1
  Firmware Version:	16,16
  Port (Upstream):
  Status:	Device connected
  Link Status:	0x2
  Speed:	Up to 40Gb/s x1
  Current Link Width:	0x2
  Link Controller Firmware Version:	0.25.0
  Port:
  Status:	Device connected
  Link Status:	0x2
  Speed:	Up to 40Gb/s x1
  Current Link Width:	0x2
  Link Controller Firmware Version:	0.25.0

Dell Thunderbolt Dock:

  Vendor Name:	Dell
  Device Name:	Dell Thunderbolt Dock
  Vendor ID:	0xD4
  Device ID:	0xB054
  Device Revision:	0x1
  UID:	0x00D45FB8F859A200
  Route String:	301
  Firmware Version:	27,27
  Port (Upstream):
  Status:	Device connected
  Link Status:	0x2
  Speed:	Up to 40Gb/s x1
  Current Link Width:	0x2
  Link Controller Firmware Version:	0.36.0
  Port:
  Status:	No device connected
  Link Status:	0x7
  Speed:	Up to 40Gb/s x1
  Current Link Width:	0x1
  Link Controller Firmware Version:	0.36.0

I have newer FW in the DOCK as well as in the MAC side. Try to do incremental update on the DOCK.
start with the 1.0.0 then with the 1.0.2.

It took me like 5 tries till I managed to successfully install the 1.0.2 dock FW.

**Comment by slingshotpeace** [comment]
Aha, @csitarichie 's dock has fw 27,27 (weird it's a comma separator), while my dock is 16.16.  I've run the TB16 firmware update utility (1.0.0 and 1.0.2) dozens and dozens of times, with various versions of Thunderbolt Controller software.  It never recognizes a dock is present to be upgraded.  The Thunderbolt software lists the TB16 and I've set the devices to Always Allow connection.  I strongly suspect the TB16 firmware update utility is incompatible the Inspiron 15 7577.

**Comment by csitarichie** [comment]
@slingshotpeace That could be I have been using a Dell Precision 5520 with windows 10 to do the update.

**Comment by peacechen** [comment]
Aha, @csitarichie 's dock has fw 27,27 (weird it's a comma separator), while my dock is 16.16. I've run the TB16 firmware update utility (1.0.0 and 1.0.2) dozens and dozens of times, with various versions of Thunderbolt Controller software. It never recognizes a dock is present to be upgraded. The Thunderbolt software lists the TB16 and I've set the devices to Always Allow connection. I strongly suspect the TB16 firmware update utility is incompatible the Inspiron 15 7577.

Thanks for posting your dock info 👍

**Comment by forroden23** [comment]
Hi all I have a MacBook Pro (15-inch, 2017) updated to Mojave 10.14.3 and the dock is recognized with out installing the thunderbolt 3, but none of the ports works. If I install the thunderbolt 3 will they work?
also dock have the latest firmware

**Comment by woohsian** [comment]
@forroden23 I am on Mojave 10.14.4 and latest TB16 firmware.
With this kext loaded - USB A ports, audio, ethernet and one display is working (mini DP) on the TB16. 
I am also using the TB16 as the power source.

**Comment by woohsian** [comment]
@forroden23 Forgot to say im using MacBookPro (15-Inch, 2018)

**Comment by peacechen** [comment]
@woohsian Do the HDMI or full size DP ports work?  Or do you mean that only the mini-DP outputs video?

@forroden23  Installing TB3Unblocker should get the USB ports working at the least.

**Comment by woohsian** [comment]
@peacechen I could not get the full size DP port to work for some reason, only mini-DP.

**Comment by yjhuoh** [comment]
using tb3unlocker, after firmware update to 1.0.2 (`DellDockingStationFwUp_1.0.2_12052018_TB16.exe`) can confirm thunderbolt 3, mini displayport, and hdmi are working for display out on my tb16. i am able to output 4k @ 60hz using the thunderbolt3, only 4k@30hz using hdmi. 

usb ports working fine. did not test sound / network card.

**Comment by krautboy** [comment]
I have the dock working on a Macbook Pro 2016. I'm running a Dell P2715Q via HDMI. It looks like it's working at 4K 60hz. I'm still experimenting with the DisplayPorts. Sound, network, USB, all work. I have a Sabrent THE-3WEA THB3 to 10Gbe Ethernet adapter that I'd like to plug into the rear TB port but it doesn't seem to work on the Mac. The lights come on for a few seconds then it disappears. Works fine on Windows. If I could just get that to work! As is, I have to plug that in directly to the Mac. I've tried a USB-C to HDMI adapter off the back and that works. VGA port works in Windows...

**Comment by fenneclabs** [comment]
@yjhuoh, what Macbook do you have?

**Comment by krautboy** [comment]
2016 Macbook Pro 13" Retina.

**Comment by yjhuoh** [comment]
MacBook Pro (13-inch, 2017, Two Thunderbolt 3 ports)

**Comment by BrendonSled** [comment]
Anyone on Catalina? I'm wondering if I should upgrade or not

**Comment by ThinkCode** [comment]
2018 MacBook Pro, Dell TB Dock, Mac OS Catalina 10.15 - HP Omen monitor w/ HDMI works, charging works.

**Comment by augustozanellato** [comment]
@ThinkCode Can you try if the rear usb2.0 are working?

**Comment by krautboy** [comment]
I just upgraded my Macbook Pro 2016 and can confirm it does work. I didn't even have to reinstall the unblocker. Also the rear USB port has a mouse plugged into it and it works. Thanks to @ThinkCode for giving me the confidence to try.

**Comment by augustozanellato** [comment]
Ok lol it seems to be a problem with my hackintosh then

**Comment by BrendonSled** [comment]
Thanks ya'll, I also upgraded and everything is working fine. Using usb-c to power a 5k display.

**Comment by kopwei** [comment]
I have 2 Macbook Pros, one is MBP 13 (2017, 4 tb3 ports) and the other one is MBP 15 (2019). I formatted the HDD and re-installed the OS to Catalina on both MBPs and then installed the unblocker.
USB ports and LAN ports works perfect.
After upgraded TB16 to latest firmware, minidisplay port start working, but not the full-sized display port.
Strange thing is, after I connected the dock to Dell U2711 (2560x1440) display with MiniDP to DP cable, MBP 15 can correctly set the displays resolution, but MBP 13 can only set the display to 1080p. 
Another issue is, if I restart MBP, the external display connected to TB16 won't come up. I have to unplug/plug the TB3 cable to make the display light up.

**Comment by micahqcade** [comment]
I have a 2018 MBP 15 with touchbar, network, audio, video and usb ports work under Catalina:
  Model Name:	MacBook Pro
  Model Identifier:	MacBookPro15,1
  Processor Name:	6-Core Intel Core i7
  Processor Speed:	2.6 GHz
  Number of Processors:	1
  Total Number of Cores:	6
  L2 Cache (per Core):	256 KB
  L3 Cache:	9 MB
  Hyper-Threading Technology:	Enabled
  Memory:	16 GB
  System Version:	macOS 10.15 (19A602)
  Kernel Version:	Darwin 19.0.0
  Boot Volume:	Macintosh HD
  Boot Mode:	Normal

I am using a TB16 running 1.02

**Comment by dude1818** [comment]
I have a 2019 MBP 15 runnning 10.15 (19A602)

TB16 charges, but network/usb not detected.

The dock cannot be detected in bootcamp, when I tried to update the firmware.

any ideas?

**Comment by micahqcade** [comment]
You can only update the firmware using a dell laptop  with thunderbolt
IIRC, sorry :(

**Comment by qnox81** [comment]
anybody having ethernet / lan port working on TB16 with Catalina 10.15/10.15.1? Mine shows port up, but no IP connectivity

**Comment by krautboy** [comment]
Built-in Ethernet works for me with Catalina. I have a Sabrent tb3 to 10gbe adapter that almost works when plugged into the rear tb3 port. Lights come on for a couple seconds... Works in Windows.

**Comment by qnox81** [comment]
did you installed any extra drivers for build int TB16 Ethernet port?

**Comment by jtreminio** [comment]
On new MBP 16", Catalina. Followed directions for TB16. Works well.

Sometimes when first connecting dock to laptop, need to remove and reconnect for MBP to recognize the dock as a power source (to charge the laptop).

Full-size DP port does not work for display-out, I need to use the dock's mini-DP port. Ethernet and USB work fine (USB devices recognized as USB 2, I need to test with a USB3 drive to make sure I get full speed).

Thanks so much for this great tool.

**Comment by TheJMan211** [comment]
My wife got a Macbook Air recently and I am using a TB16 dock. The Mac has the latest update, 10.15.2. I'm seeing behavior that it seems is different than what others describe. First, it never showed "Unsupported" for the dock in the System Information, but nothing worked anyway besides charging. I updated the dock firmware and now if I reboot the Mac with the docking station attached I get display on my DP monitor that I can extend or mirror but if the computer goes to sleep or turns off displays or is unplugged from the dock and plugged back in the display does not come back. Rebooting is the only way it comes back. In addition, none of the USB ports seem to work. Not using ethernet so I can't test that. I tried installing the unblocker anyway to see if it would get the USB working but I can't tell if it installed since it never said that it was unsupported anyway but the USB ports still don't work. I don't have an HDMI monitor to test with to find out if HDMI would work better than the DP.

**Comment by threadripper** [comment]
Also have this problem in my TB15 cable, USB2.0/3.0 RJ56 mic work well, except video output.

**Comment by dnasir** [comment]
List of things that work on my 2019 MacBook Pro 16" on the TB16:

* Display connected to the Thunderbolt port
* USB
* Ethernet

I haven't tested the other display ports because I have no need for them.

**Comment by tmerten** [comment]
Here is what I found with a TB16 and MBP 13" 2019:

* USB working
* Ethernet working
* Sonnet EGPU via Thunderbolt port (daisy chain) working (although might be slower than direct connect)

However, I also have a weird issue:
 
* After the firmware update on the TB16, when I now connect it to OSX, the Mac detects an additional, non existing display which is shown as `Non`. Anybody seen this?

**Comment by tinhtruong** [comment]
After upgrading the firmware of the dock to the latest version 1.0.4 which can be downloaded at https://www.dell.com/support/home/en-ng/drivers/driversdetails?driverid=fp58v using a Dell laptop on Windows 10
My machine: MacBook Pro (15-inch, 2018) with Touchbar - macOS Catalina 10.15.5
Thunderbolt3Unblocker: v0.0.4
What works:
- HDMI port
- Thunderbolt port (USB-C) at the back
- MiniDisplayPort (partially, there are lines running across my Dell monitor)
- Audio output
- All USB ports
- Charging
What DOES NOT work:
- Displayport 

I have not tested the Ethernet port.
I can have 2 monitors hooking up with the dock (one from HDMI and one from the Thunderbolt(USB-C) at the back) working. If I open the laptop screen then I have 3 screen working (no mirroring)
I hope this will help someone out there searching for a solution like me 1 month ago :).

**Comment by tinhtruong** [comment]
Just upgraded my macbook pro to Catalina 10.15.16 and the dock still works as expected.

**Comment by mr-perseus** [comment]
I managed to get it to work with two displays, LAN, power, USB keyboard and mouse:

- Update the dock's driver (see above).
- Install Thunderbolt3Unblocker on macOS
- Use MiniDisplayPort (or HDMI) and Thunderbolt 3 for the displays. VGA and DisplayPort didn't work for me. You can't use MiniDisplayPort and HDMI because they are daisy chained (not supported by macOS).

I also installed the DisplayLink driver on macOS (brew cask install displaylink) before enabling Thunderbolt3Unblocker. However, I'm uncertain if this is necessary.

**Comment by dnasir** [comment]
I recently noticed that the charging is a bit unstable at times, and there have been several instances where my MBP16 is simply not charged while connected to the dock. I plug it in, I see the thunderbolt symbol in the battery icon in the top right hand corner, but when I click on it, it tells me that it's not charging.

A system restart solves this problem.

Also, it's charging at just 60W according to System Report, but I know this dock is capable of doing 120W, because I've used it with an XPS15 in the past. I'm guessing that there's some sort of low level flag that enables 120W power delivery when it's connected to a Dell laptop.

**Comment by orappold** [comment]
Yes, same happens to me - re-pluging does work for me without restarting. 
Also I do have issues that sometimes my Mac crashes completely. I then need to unplug it and boot it manually, then plug it again. Anyone having the same issue?

**Comment by micahqcade** [comment]
If/when you update to macOS Catalina 10.15.6 you have disable SIP again, rerun the command

"
sudo sh -c 'chown -R root:wheel /Library/Extensions/Thunderbolt3Unblocker.kext && \
chmod -R g-w,o-w /Library/Extensions/Thunderbolt3Unblocker.kext && \
kextcache -i / && \
nvram -d t3u-incompatible && \
kextload -b es.govost.ryan.Thunderbolt3Unblocker'
"
and reboot, then the dock should work. I just updated to 10.15.6 and the kext stopped working even though it was still in the 
Library/extensions folder

**Comment by peacechen** [comment]
☝️  That should be added to the Readme.

**Comment by ioExpander** [comment]
Hi. I have a 2020 13 inch Macbook Pro with 4 TB ports. It's running Catalina 10.15.6. 
My TB16 dock is running 1.0.4 flashed through a Dell PC running windows. I can get USB ports working fine, however the HDMI or Mini Display port do not output anything to my external 4k monitor (from Acer). I've tried downgrading the dock to 1.0.2 with no success... and now running out of ideas 😞 
Did anyone with a similar configuration manage to get any video out signal ?

Here is the output config from the dock : 
> system_profiler SPThunderboltDataType



    Thunderbolt Bus 1:

      Vendor Name: Apple Inc.
      Device Name: MacBook Pro
      UID: 0x4C374C4D19030011
      Route String: 0
      Firmware Version: 0,0
      Domain UUID: FA3C6DE8-11FD-4A5A-B573-715468623053
      Port:
          Status: Device connected
          Link Status: 0x2
          Speed: Up to 40 Gb/s x1
          Current Link Width: 0x2
          Receptacle: 0
          Link Controller Firmware Version: 1.38.0
      Port:
          Status: No device connected
          Link Status: 0x7
          Speed: Up to 40 Gb/s x1
          Current Link Width: 0x1
          Receptacle: 0
          Link Controller Firmware Version: 1.38.0

        Dell Thunderbolt Cable:

          Vendor Name: Dell
          Device Name: Dell Thunderbolt Cable
          Vendor ID: 0xD4
          Device ID: 0xB051
          Device Revision: 0x1
          UID: 0x80865F31B489C910
          Route String: 1
          Firmware Version: 16,16
          Port (Upstream):
              Status: Device connected
              Link Status: 0x2
              Speed: Up to 40 Gb/s x1
              Current Link Width: 0x2
              Link Controller Firmware Version: 0.25.0
          Port:
              Status: Device connected
              Link Status: 0x2
              Speed: Up to 40 Gb/s x1
              Current Link Width: 0x2
              Link Controller Firmware Version: 0.25.0

            Dell Thunderbolt Dock:

              Vendor Name: Dell
              Device Name: Dell Thunderbolt Dock
              Vendor ID: 0xD4
              Device ID: 0xB054
              Device Revision: 0x1
              UID: 0x00D45F31E0F04300
              Route String: 301
              Firmware Version: 27,27
              Port (Upstream):
                  Status: Device connected
                  Link Status: 0x2
                  Speed: Up to 40 Gb/s x1
                  Current Link Width: 0x2
                  Link Controller Firmware Version: 0.36.0
              Port:
                  Status: No device connected
                  Link Status: 0x7
                  Speed: Up to 40 Gb/s x1
                  Current Link Width: 0x1
                  Link Controller Firmware Version: 0.36.0

**Comment by snowspirit** [comment]
> sudo sh -c 'chown -R root:wheel /Library/Extensions/Thunderbolt3Unblocker.kext &&
> chmod -R g-w,o-w /Library/Extensions/Thunderbolt3Unblocker.kext &&
> kextcache -i / &&
> nvram -d t3u-incompatible &&
> kextload -b es.govost.ryan.Thunderbolt3Unblocker'

Thank you for this! This should be in the Read me for sure as it's what got it to work on Catalina 10.15.6

**Comment by ThinkCode** [comment]
Just checking to see if anything has changed with dual monitor support (extend mode). It is glitchy and is simply not usable in its current form. Wondering if anyone else had better success.

**Comment by AdaLollA** [comment]
I just updated my TB16 with an XPS15 in an attempt to get a connection to a 2019 MacBook Pro. 

I followed the tb3unblockers instructions to the best of my abilities and i am pretty sure all should be good on that side.

The problem is that none of the Thunderbolt Buses show a device connection. They both are showing "no device connected". The MacBook is charging, but none of the ports on the Dock are working.

Anybody have an idea what is going on / what I did wrong? Seems to me that the device isn't even attempting to use TB3? To the MacBook it seems to just be a "charging cable"?

I get the following system information from the system report:

>   Port:
>   Status:	**No device connected**
>   Link Status:	0x101
>   Speed:	Up to 40Gb/s x1
>   Current Link Width:	0x1
>   Receptacle:	2
>   Link Controller Firmware Version:	1.41.0
>   Port:
>   Status:	**No device connected**
>   Link Status:	0x101
>   Speed:	Up to 40Gb/s x1
>   Current Link Width:	0x1
>   Receptacle:	1
>   Link Controller Firmware Version:	1.41.0

**Comment by jackie099** [comment]
I am so close to getting it working. I bought a Dell XPS15 to upgrade the firmware of the dock to 1.0.4...
Got the double monitor setup working with XPS15 after the firmware upgrade. 

Here is the monitor setup: 
1. 4k 60hz monitor -> usb-c to HDMI 2.0 cable connecting to tb3 port of tb16
2. 1440p 60hz monitor -> HDMI to HDMI cable connecting to tb3 port of tb16

I have a MacBook Pro 16 2019, running Catalina 10.15.7. Both monitors work after connecting to MacBook Pro 16, but the 1440p monitor flickers every 5 seconds... 

I've tried to reset smc/nvram, reinstall displaylink driver, play with different resolution/refresh rate, trying different tb3 port, and had no luck. 

I gave a try with my old MacBook Pro 15 2018 at the end, and it just works... without flickering or anything. I quickly Googled and found someone has the same issue with TB16 and mbp16.  https://apple.stackexchange.com/questions/379778/macbook-pro-16-and-dell-tb16

Anyone experiencing the same with a solution?

**Comment by jackie099** [comment]
It seems the issue of blinking/flickering on Macbook Pro 16 only is caused by CPU/USB driver: https://www.dell.com/support/article/de-de/sln306731/kaby-lake-systems-with-usb-type-c-dp-only-non-thunderbolt-connections-using-wd15-or-ds1000-may-have-blanking-external-video?lang=en

I don't see a solution on macOS yet, anyone?

**Comment by jackie099** [comment]
![20201118_175956](https://user-images.githubusercontent.com/7269348/99599195-d8a7b680-29c8-11eb-9e1a-2a4dd806cf20.jpg)

Another quick update, I just tried to install windows 10 with bootcamp, and guess what... It works out of the box and no driver install required. It's definitely macOS + MacBook Pro 16 issues combined...

**Comment by tanpengsccd** [comment]
> Just FYI folks
> 
> The Dell D6000 docks work with the Macbook using DisplayLink software. I have one for work and one for home. Home uses Dual 4k and at work I'm using triple 1080p monitors. No issues plus it charges the macbook.
> https://www.dell.com/en-us/shop/dell-universal-dock-d6000/apd/452-bcyt/pc-accessories

Dose  BigSur work too?

**Comment by mr-perseus** [comment]
My setup still works on Big Sur after reinstalling this kernel extension:

- MacBookPro14,2 (macOS 11.0.1)
- Updated driver on Dell TB16 (see OP)
- Display 1 on MiniDisplayPort (or HDMI), Display 2 on Thunderbolt 3. (DisplayPort and VGA don't work for me, MiniDisplayPort and HDMI are daisy chained so you can only use one of them)

The DisplayLink driver was not necessary for me.

**Comment by Kostya950** [comment]
Please help, after update to macOs 11.1 I can't build project with Xcode, I have got the message that build success but Thunderbolt3Unblocker.kext doesn't appear. 
What am I doing wrong?

**Comment by Kostya950** [comment]
I have problem with macOs 11.1 - doesn't work for me...

**Comment by mrkita** [comment]
I was using this more or less fine for 2 years on my macbook pro 2016 model. I would occasionally not charge, but replugging it solved the problem, and I could use the usb-c/thunderbolt with a hdmi convertor cable to my monitor without problems.

Flash forward and I have a macbook 2019 16" pro model, and it just doesn't charge correctly. It always says "not charging", and drains directly. I checked the system report as mentioned above, and indeed it only lists it as a 60w charge, but also says "not charging".

Anyone have any clues on this? I'm getting really sick of this dock...

**Comment by jackie099** [comment]
> I was using this more or less fine for 2 years on my macbook pro 2016 model. I would occasionally not charge, but replugging it solved the problem, and I could use the usb-c/thunderbolt with a hdmi convertor cable to my monitor without problems.
> 
> Flash forward and I have a macbook 2019 16" pro model, and it just doesn't charge correctly. It always says "not charging", and drains directly. I checked the system report as mentioned above, and indeed it only lists it as a 60w charge, but also says "not charging".
> 
> Anyone have any clues on this? I'm getting really sick of this dock...

MacBook Pro 16 draws as much as 100 watts under full load. Using external monitors will trigger the usage of discrete graphics, which easily causes the whole system to draw more than 60 watts...

**Comment by alexchandel** [comment]
Anyone have any recent success with TB16 + video?

**Comment by Sicmatr1x** [comment]
> Anyone have any recent success with TB16 + video?

nop, just USB success, and if you connect your time machine backup harddrive on the TB16, it couldn't work fine and hint could not begin to backup because your mac is battery mode, but my mac is charging, seems it could not recognize charging

**Comment by laris** [comment]
I have a tb16 dock and tried to upgrade firmware to 1.0.5 (latest version in 2022 April) but only Dock NVM firmware failed to upgrade to 0.0.27. I'm trying to get Dell support to fix this.

Now the dock can work with usb type-c to 4k monitor, but system cannot show it, and sometimes usb can work and sometimes usb cannot work.


```
kextstat|grep -i unblocker
Executing: /usr/bin/kmutil showloaded
No variant specified, falling back to release
  210    0 0xffffff7f9a5bf000 0x4e000    0x4e000    es.govost.ryan.Thunderbolt3Unblocker (1) E6321B23-449C-366C-A128-F6B88439C8C0 <209 84 9 7 6 3>
```

```
system_profiler SPSoftwareDataType SPHardwareDataType SPThunderboltDataType
Software:

    System Software Overview:

      System Version: macOS 12.3.1 (21E258)
      Kernel Version: Darwin 21.4.0
      Boot Volume: MacBookPro
      Boot Mode: Normal
      User Name: Laris (laris)
      Secure Virtual Memory: Enabled
      System Integrity Protection: Disabled
      Time since boot: 7:15

Hardware:

    Hardware Overview:

      Model Name: MacBook Pro
      Model Identifier: MacBookPro16,4
      Processor Name: 8-Core Intel Core i7
      Processor Speed: 2.3 GHz
      Number of Processors: 1
      Total Number of Cores: 8
      L2 Cache (per Core): 256 KB
      L3 Cache: 16 MB
      Hyper-Threading Technology: Enabled
      Memory: 32 GB
 

Thunderbolt/USB4:

      Thunderbolt/USB4: No drivers are loaded.
```

**Comment by Tuan2002** [comment]
I updated to lastest fw 1.0.5 but video out don't work... Anyone help me!. macos 12.6

**Comment by AlexMohanMD** [comment]
Hi, to anyone struggling with this, with the latest fw and unlocker for ventura, you can use a cheap usb-c to hdmi adapter which works perfectly without flickering! just connect it to the daisy-chain port

**Comment by conanslash** [comment]
what the latest firmware and how can i get it? from dell official website, the latest releases is 2019...

**Comment by smastilovich** [comment]
I have a Macbook Pro 13" 2017 with latest version of Ventura 13.5, as well as a TB16 dock with latest firmware. Anyone know where I can get the "unlocker for Ventura"? The latest version of Thunderbolt3Unblocker is from 2018?
